client/lib/docker.ts

```diff
@@ -0,0 +1,160 @@
+//const tar = require('tar-fs')
+//const Docker = require('dockerode');
+//const docker = new Docker({socketPath: '/var/run/docker.sock'});
+//const getPort = require('get-port');
+
+import { existsSync, createWriteStream } from 'fs'
+import * as Docker from 'dockerode';
+import * as tar from 'tar-fs';
+
+export class ContainerInteractor{
+	docker: Docker;
+	constructor() { 
+		this.docker = new Docker({socketPath: '/var/run/docker.sock'});
+	}
+	
+	async createCoreWorker(ports: number[]) {
+		let corePort = ports[0];
+		let workerPort = ports[1];
+		
+		let coreOpts = {
+			Image: '',
+			OpenStdin: true,
+			Env: [
+				`UDEV=0`,
+				`WORKER_TYPE=qemu`,
+			],
+			AttachStdout: true,
+			AttachStderr: true,
+			HostConfig: {
+				Privileged: true,
+				NetworkMode: `host`,
+				Mounts: [
+					{
+						Target:   "/data",
+						Source:   "core-storage",
+						Type:     "volume", 
+						ReadOnly: false
+					},
+					{
+						Target:   "/reports",
+						Source:   "reports",
+						Type:     "volume", 
+						ReadOnly: false
+					}
+				]
+			}
+		}
+
+		let workerOpts = {
+			Image: '',
+			OpenStdin: true,
+			Env: [
+				`UDEV=0`,
+				`WORKER_TYPE=qemu`,
+			],
+			AttachStdout: true,
+			AttachStderr: true,
+			Devices:[
+				{
+					PathOnHost: "/dev/net/tun",
+					PathInContainer: "/dev/net/tun"
+				},
+				{
+					PathOnHost: "/dev/kvm",
+					PathInContainer: "/dev/kvm"
+				}
+			],
+			CapAdd: [
+				"NET_ADMIN"
+			],
+			HostConfig: {
+				Privileged: true,
+				NetworkMode: `host`,
+				Mounts: [
+					{
+						Target:   "/data",
+						Source:   "core-storage",
+						Type:     "volume", 
+						ReadOnly: false
+					},
+					{
+						Target:   "/reports",
+						Source:   "reports",
+						Type:     "volume", 
+						ReadOnly: false
+					}
+				]
+			}
+		}
+
+		console.log(`Found unused ports: ${corePort}, ${workerPort}`);
+		
+		let env = [
+			`UDEV=0`,
+			`WORKER_TYPE=qemu`,
+			`WORKER_PORT=${workerPort}`,
+			`CORE_PORT=${corePort}`,
+			`SCREEN_CAPTURE=true`
+		]
+
+		coreOpts.Env = env;
+		workerOpts.Env = env;
+		await this.docker.createVolume(`core-storage`)
+		await this.docker.createVolume(`reports`)
```

Comment:
These volumes aren't namespaced and will conflict with other instances on the same host.

We also want to clean them up when we are done with them. Or even better avoid needing volumes at all.